packages/webpack/src/resource_loader.ts

```diff
@@ -0,0 +1,116 @@
+import {ResourceLoader} from '@angular/compiler';
+import {readFileSync} from 'fs';
+import * as vm from 'vm';
+import * as path from 'path';
+
+const NodeTemplatePlugin = require('webpack/lib/node/NodeTemplatePlugin');
+const NodeTargetPlugin = require('webpack/lib/node/NodeTargetPlugin');
+const LoaderTargetPlugin = require('webpack/lib/LoaderTargetPlugin');
+const SingleEntryPlugin = require('webpack/lib/SingleEntryPlugin');
+
+
+
+export class WebpackResourceLoader implements ResourceLoader {
```

Comment:
Since I haven't looked at `ResourceLoader` I don't really know, but since this replaces angular2-template-loader and family, we probably are going to end up with some sad folks because they can't use sass/less/styl inside of components. But, at the cost of AoT, first pass with room to refactor, its justifiable. Just means there is a lot more responsibility on the ResourceLoader to handle all these things.